matsubara/Chap8/knock73.py

```diff
@@ -0,0 +1,122 @@
+import os
+import knock71
+import knock72
+import numpy as np
+import math
+from sklearn.model_selection import train_test_split
+
+
+def read_file(filepath: str):
+    with open(filepath, "r", encoding="utf-8") as f:
+        text = f.read()
+        words = text.split()
+    return words
+
+
+def write_file(filepath: str, word_list: list):
+    with open(filepath, "w", encoding="utf-8") as f:
+        f.write(" ".join(word_list))
+
+
+def bow_vector(words: list, standard_words: list):
+    vector = np.zeros(len(standard_words), dtype="float32")
+    for word in words:
+        if word in standard_words:
+            vector[standard_words.index(word)] += 1
+    return vector
+
+
+def input_filelines(filepath: str):
+    with open(filepath, "r", encoding="cp1252") as f:
+        return f.readlines()
+
+
+def make_vector_and_label(sent_lines: list, adj_list: list, stop_word: list):
+    sent_adj = []
+    labels = []
+    for line in sent_lines:
+        sent_adj.append(knock72.pull_adj_bow(line, stop_word, True))
+        if line[:2] == "+1":
+            labels.append(1)
+        elif line[:2] == "-1":
+            labels.append(-1)
+        else:
+            continue
+    vector = [bow_vector(sent, adj_list) for sent in sent_adj]
+    labels = np.array(labels)
+    return vector, labels
+
+
+def discriminant_function(y: int, W: list, f: list):
```

Comment:
W と f はnumpyベクトルでないっけ？
```suggestion
def discriminant_function(y: int, W: np.ndarray, f: np.ndarray):
```